src/bidiMapper/domains/network/NetworkProcessor.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  type Network,
+  type EmptyResult,
+  UnknownCommandException,
+  NoSuchInterceptException,
+} from '../../../protocol/protocol.js';
+import {uuidv4} from '../../../utils/uuid.js';
+
+import type {NetworkStorage} from './NetworkStorage.js';
+
+export class NetworkProcessor {
+  readonly #networkStorage: NetworkStorage;
+
+  constructor(networkStorage: NetworkStorage) {
+    this.#networkStorage = networkStorage;
+  }
+
+  addIntercept(
+    params: Network.AddInterceptParameters
+  ): Network.AddInterceptResult {
+    const intercept = uuidv4();
+
+    const urlPatterns: string[] = params.urlPatterns ?? [];
+    const parsedPatterns: string[] = [];
+    for (const urlPattern of urlPatterns) {
+      const parsed = urlPattern; // TODO: Parse the pattern.
+      parsedPatterns.push(parsed);
+    }
+
+    // TODO: call CDP `Fetch.enable`.
+
+    this.#networkStorage.interceptMap.set(intercept, {
```

Comment:
let's not expose maps directly, it would be really beneficial if we wrap map access